data-prepper-pipeline-parser/src/test/java/org/opensearch/dataprepper/pipeline/parser/EnumDeserializerTest.java

```diff
@@ -0,0 +1,166 @@
+package org.opensearch.dataprepper.pipeline.parser;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.BeanProperty;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.TextNode;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.opensearch.dataprepper.model.event.HandleFailedEventsOption;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class EnumDeserializerTest {
+
+    private ObjectMapper objectMapper;
+
+    @BeforeEach
+    void setup() {
+        objectMapper = mock(ObjectMapper.class);
+    }
+
+    private EnumDeserializer createObjectUnderTest(final Class<?> enumClass) {
+       return new EnumDeserializer(enumClass);
+    }
+
+    @Test
+    void non_enum_class_throws_IllegalArgumentException() {
+        assertThrows(IllegalArgumentException.class, () -> new EnumDeserializer(Duration.class));
+    }
+
+    @ParameterizedTest
+    @EnumSource(HandleFailedEventsOption.class)
```

Comment:
Let's rely on test classes to avoid changes in one class affecting others. Maybe update `TestEnum` with more values.